org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -0,0 +1,356 @@
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,          *
+* MA 02110-1301, USA.                                                 *
+*                                                                     *
+* Contributors:                                                       *
+* - Igor Pojzl, Cloudempiere                                          *
+* - Peter Takacs, Cloudempiere                                        *
+**********************************************************************/
+package org.adempiere.webui.editor;
+
+import java.beans.PropertyChangeEvent;
+import java.util.Objects;
+
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.util.GridRowCtx;
+import org.adempiere.webui.ValuePreference;
+import org.adempiere.webui.apps.AEnv;
+import org.adempiere.webui.component.Textbox;
+import org.adempiere.webui.component.ToolBarButton;
+import org.adempiere.webui.event.ContextMenuEvent;
+import org.adempiere.webui.event.ContextMenuListener;
+import org.adempiere.webui.event.ValueChangeEvent;
+import org.adempiere.webui.theme.ThemeManager;
+import org.adempiere.webui.window.WFieldRecordInfo;
+import org.adempiere.webui.window.WRecordIDDialog;
+import org.compiere.model.GridField;
+import org.compiere.model.MColumn;
+import org.compiere.model.MLookup;
+import org.compiere.model.MLookupFactory;
+import org.compiere.model.MLookupInfo;
+import org.compiere.model.MRole;
+import org.compiere.model.MTable;
+import org.compiere.util.DisplayType;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+import org.compiere.util.Util;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+
+/**
+ * 
+ * @author Peter Takacs, Cloudempiere
+ *
+ */
+public class WRecordIDEditor extends WEditor implements ContextMenuListener, IZoomableEditor {
+	
+	/** Is Read/Write enabled on the editor */
+	private boolean m_ReadWrite;
+	/** Old value (Record_ID) */
+	private Object recordIDValue;
+	/** Old value (AD_Table_ID) */
+	private Object tableIDValue;
+	
+	/** Current tab's AD_Table_ID GrodField */
+	private GridField tableIDGridField;
+	
+	// UI components
+	private Textbox recordTextBox;
+	private ToolBarButton editButton;
+	private ToolBarButton zoomButton;
+	
+	// images
+	private static final String IMAGES_CONTEXT_ZOOM_PNG = "images/Zoom16.png";
+	private static final String IMAGES_CONTEXT_EDIT_RECORD_PNG = "images/EditRecord16.png";
+
+	/**
+	 * Constructor
+	 * @param gridField
+	 * @param tableEditor
+	 * @param editorConfiguration
+	 */
+	public WRecordIDEditor(GridField gridField, boolean tableEditor, IEditorConfiguration editorConfiguration) {
+		super(new Div(), gridField, tableEditor, editorConfiguration);
+		
+		getComponent().setSclass("recordid-editor");
+		getComponent().addEventListener(Events.ON_RIGHT_CLICK, this);
+		
+		init();
+	}
+
+	private void init() {
+		if(gridTab == null) {
+			throw new IllegalArgumentException("gridTab is null");
```

Comment:
gridTab is null when we call this from FindWindow, so now the find window cannot be opened on windows with the RecordID editor because this exception is thrown. If you suggest to have this exception here, we should handle the gridTab == null situation somewhere. Maybe ignore this editor in FindWindow.java by adding a check to addSelectionColumn(GridField mField, Group group) method?

```
if(DisplayType.RecordID == mField.getDisplayType())
      	return false;
```

Reply:
does it works on the find window if we remove the gridtab exception ?

Reply:
yes, it works if we'd add a simple `return;` statement instead of the exception

Reply:
alright, if it is working fine, you can roll it back to the simple return.

Reply:
hmm .. just tested and it seems you can't use the record id editor at find window. what do you means by it is working ?

Reply:
Oh sorry I misunderstood you, taught we are talking about skipping the editor in find window... I can get it work by sending the GridTab from Find Window. Providing the fix...

Reply:
So after further analysis, it looks like Find Window doesn't have its own GridTab, so I can't access the AD_Table_ID field of the Find Window. If I'm correct, through the GridTab in FindWindow I can only access the AD_Table_ID of the Window itself (that would cause that I can change AD_Table_ID of the Window from FindWindow)